### ISSUE TITLE
AL: PHL Orders Staging Connectivity

### ISSUE THREAD

**Issue by halprin** [CDCgov/trusted-intermediary#670]
# Story
As a AL (Natus) Lab, so that AL can receive NBS ETOR orders, I need a sample order to show up in my integration engine.

## Pre-conditions
- We have access to or have access to someone with access to the integration engine.
- We can format/send a baseline OML message.  
- Additional stories will be created to make the rest of the order show up correctly. 

## Acceptance Criteria
- [ ] Sample order can be parsed correctly and available in the integration engine.

###  Engineering

- [x] Send ReportStream Programmers Guide, a postman collection, and details on how to be an RS receiver to Neelima of AL PHL. - @basiliskus 
- [x] Sent AL PHL host, username, and credentials for SFTP

## Definition of Done
- [ ] Documentation tasks completed
  - [ ] Documentation and diagrams created or updated
    - [ ] ADRs (`/adr` folder)
    - [ ] Main `README.md`
    - [ ] Other READMEs in the repo
    - [ ] If applicable, update the `ReportStream Setup` section in `README.md`
  - [ ] Threat model updated
  - [ ] API documentation updated
  - [ ] Source code documentation created when the code is not self-documenting
- [ ] Code quality tasks completed
  - [ ] Logging added where useful
  - [ ] Code refactored for clarity and no design/technical debt
  - [ ] Adhere to separation of concerns; code is not tightly coupled, especially to 3rd party dependencies
  - [ ] Code is reviewed or developed by pair; 1 approval is needed but consider requiring an outside-the-pair reviewer
  - [ ] Code quality checks passed
- [ ] Security & Privacy tasks completed
  - [ ] Security & privacy gates passed
- [ ] Testing tasks completed
  - [ ] Load tests passed
  - [ ] Unit test coverage of our code >= 90%
- [ ] Build & Deploy tasks completed
  - [ ] Build process updated
  - [ ] API(s) are versioned
  - [ ] Feature toggles created and/or deleted.  Document the feature toggle
  - [ ] Source code is merged to the main branch

## Notes

- Depends on: https://github.com/CDCgov/prime-reportstream/issues/12714

**Comment by basiliskus** [comment]
Sent an email to Neelima and AL-PHL folks on 11/21 (under subject "Re: Question about REST API") with the REST API requirements from ReportStream to set up a receiver (+ the Programmers Guide and a postman collection with test connection settings). We're waiting for a response to continue helping AL with onboarding to RS

**Comment by basiliskus** [comment]
Started a conversation with Neelima and Ott on keybase and sent her the API key

**Comment by basiliskus** [comment]
Neelima tried unsuccessfully to send a message to the reports API. She tried using Rhapsody and curl. I found some issues with the curl command she used and sent her a revised and successfully tested command for her to try. Waiting for a response

**Comment by basiliskus** [comment]
Neelima is working with Rhapsody's help desk to try to submit a successful request to the reports API using Rhapsody

**Comment by basiliskus** [comment]
Neelima got response from Rhapsody's help desk. The version of the HTTP client she's using doesn't support the cipher suits ReportStream supports, so she can't communicate with RS API. We reached out to RS and they created [this story](https://github.com/CDCgov/prime-reportstream/issues/12714) to explore adding support for one of the cipher suites AL-PHL Rhapsody HTTP client supports

**Comment by basiliskus** [comment]
It was noted by ReportStream that given `CBC` has flaws that make all the options weak ciphers, they will need approval fro the ISSO

**Comment by basiliskus** [comment]
Jeremy from ReportStream sent an email today to Dewayne requesting approval for one of the cipher suites

**Comment by basiliskus** [comment]
Dewayne said that the cipher suites should meet CDC's requirement, which is at least a minimum of using 256 encryption. We're now waiting for RS to confirm they will add support for one of the ciphers

**Comment by nvcllns** [comment]
Morning - based on conversation this week, I understand RS has provided confirmation that they will add support for the needed ciphers. What is the ETA on that effort so this can be communicated back to our ADPH site partners?

**Comment by sfradkin** [comment]
@nvcllns - The RS issue for this is: https://github.com/CDCgov/prime-reportstream/issues/12714 and they have indicated that this is allocated for their next sprint which runs January 17-30.

**Comment by basiliskus** [comment]
ReportStream is working this sprint on the story to add support for the ciphers. Status for the story can be followed [here](https://github.com/CDCgov/prime-reportstream/issues/12714)

**Comment by JohnNKing** [comment]
@JohnNKing Revise story; scope to integration that supports message receipt by Rhapsody; then create a new story for data visibility in LIMS

**Comment by JohnNKing** [comment]
Waiting to hear back from ADPH -- if they can support key based auth and if we need to use either their SFTP service or if they can add an scp client via a scheduled task.

**Comment by scleary1cs** [comment]
AL no longer an initial implementation partner.
